libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py

```
@@ -737,33 +385,26 @@ def retrieve_existing_index(self) -> Tuple[Optional[int], Optional[str]]:
-            params={
-                "index_name": self.index_name,
-                "node_label": self.node_label,
-                "embedding_node_property": self.embedding_node_property,
-            },
+        index_information = retrieve_vector_index_info(
+            driver=self._driver,
+            index_name=self.index_name,
+            label_or_type=self.node_label,
+            embedding_property=self.embedding_node_property,
         )
-        # sort by index_name
-        index_information = sort_by_index_name(index_information, self.index_name)
-        try:
-            self.index_name = index_information[0]["name"]
-            self.node_label = index_information[0]["labelsOrTypes"][0]
-            self.embedding_node_property = index_information[0]["properties"][0]
-            self._index_type = index_information[0]["entityType"]
-            embedding_dimension = None
-            index_config = index_information[0]["options"]["indexConfig"]
-            if "vector.dimensions" in index_config:
-                embedding_dimension = index_config["vector.dimensions"]
-
-            return embedding_dimension, index_information[0]["entityType"]
-        except IndexError:
+        if index_information:
+            try:
+                self.index_name = index_information["name"]
+                self.node_label = index_information["labelsOrTypes"][0]
+                self.embedding_node_property = index_information["properties"][0]
+                self._index_type = index_information["entityType"]
+                embedding_dimension = None
+                index_config = index_information["options"]["indexConfig"]
+                if "vector.dimensions" in index_config:
+                    embedding_dimension = index_config["vector.dimensions"]
+                return embedding_dimension, index_information["entityType"]
+            except IndexError:
+                return None, None
+        else:
             return None, None
```

I know the previous code did this, but returning `None, None` could cause silent failures. Have you considered raising an Exception here?

---

I see thanks. In that case, can we have this returned a single `None`? It seems strange that the user of `retrieve_existing_index` has to parse a tuple and check that both elements is `None`. Having a single `None` makes the function's contract more explicit.


IMO if the metadata doesn't exist then this check should be enough e.g.

```python
metadata = self.retrieve_existing_index()

if metadata:
    embedding_dim, entity_type = metadata
    ...
else:
    ...
```

instead of

```python
embedding_dim, entity_type = self.retrieve_existing_index()

if embedding_dim and entity_type:
    ...
```

Would this introduce any breaking change though?

Also ideally the function should return a dataclass/data type but it's not blocking this PR

---

Basically return `Optional[Tuple[int, str]]` instead of `Tuple[Optional[int], Optional[str]]`

---

Updated here [23ee62e](https://github.com/langchain-ai/langchain-neo4j/pull/45/commits/23ee62e3fb9887bd4f1bf3d0a63107b4779bf77d)